Libraries/LibWeb/HTML/HTMLLinkElement.cpp

```diff
@@ -596,8 +598,31 @@ void HTMLLinkElement::resource_did_load_favicon()
     document().check_favicon_after_loading_link_resource();
 }
 
-static NonnullRefPtr<Core::Promise<Web::Platform::DecodedImage>> decode_favicon(ReadonlyBytes favicon_data, URL::URL const& favicon_url, GC::Ref<DOM::Document> document)
+static NonnullRefPtr<Core::Promise<bool>> decode_favicon(ReadonlyBytes favicon_data, URL::URL const& favicon_url, GC::Ref<DOM::Document> document)
 {
+    if (favicon_url.basename().ends_with(".svg"sv)) {
```

Comment:
It might be infeasible to do but should you check for `image/svg+xml` mine type instead of file extension?
https://www.w3.org/TR/SVG11/mimereg.html